error: invalid conversion from 'int' to 'esp_cpu_watchpoint_trigger_t' (IDFGH-7712)

Since my latest esp-idf rebase I get this compiler error:

```
<firmware>/esp-idf/components/esp_system/include/esp_debug_helpers.h: In function 'esp_err_t esp_set_watchpoint(int, void*, int, int)':
<firmware>/esp-idf/components/esp_system/include/esp_debug_helpers.h:118:50: error: invalid conversion from 'int' to 'esp_cpu_watchpoint_trigger_t' [-fpermissive]
  118 |     return esp_cpu_set_watchpoint(no, adr, size, flags);
      |                                                  ^~~~~
      |                                                  |
      |                                                  int
In file included from <firmware>/esp-idf/components/esp_hw_support/include/hal/cpu_ll.h:18,
                 from <firmware>/esp-idf/components/esp_hw_support/include/hal/cpu_hal.h:17,
                 from <firmware>/esp-idf/components/esp_hw_support/include/spinlock.h:11,
                 from <firmware>/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include/freertos/portmacro.h:74,
                 from <firmware>/esp-idf/components/freertos/FreeRTOS-Kernel/include/freertos/portable.h:58,
                 from <firmware>/esp-idf/components/freertos/FreeRTOS-Kernel/include/freertos/FreeRTOS.h:70,
                 from <firmware>/esp-idf/components/esp_event/include/esp_event.h:12,
                 from <firmware>/esp-idf/components/esp_wifi/include/esp_wifi.h:56
<firmware>/esp-idf/components/esp_hw_support/include/esp_cpu.h:475:109: note:   initializing argument 4 of 'esp_err_t esp_cpu_set_watchpoint(int, const void*, size_t, esp_cpu_watchpoint_trigger_t)'
  475 | esp_err_t esp_cpu_set_watchpoint(int wp_num, const void *wp_addr, size_t size, esp_cpu_watchpoint_trigger_t trigger);
      |                                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```

I am not calling the watchpoint functionality myself.

```
[1/128 11.9/sec] Performing build step for 'bootloader'
[1/1 32.3/sec] cd /home/feedc0de/<-removed->/build/bootloader/esp-idf/esptool_py && python /home/feedc0de/<-removed->/esp-idf/components/partition_table/check_sizes.py --offset 0xf000 bootloader 0x1000 /home/feedc0de/<-removed->/build/bootloader/bootloader.bin
Bootloader binary size 0xe000 bytes. 0x0 bytes (0%) free.
[2/126 1.6/sec] Building CXX object esp-idf/espasyncudplistener/CMakeFiles/__idf_espasyncudplistener.dir/src/asyncudplistener.cpp.obj
[3/126 2.2/sec] Building CXX object esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifiutils.cpp.obj
[4/126 2.3/sec] Building CXX object esp-idf/espconfiglib/CMakeFiles/__idf_espconfiglib.dir/src/configwrapper_espwifistack.cpp.obj
[5/126 2.6/sec] Building CXX object esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/udpsender.cpp.obj
[6/126 2.5/sec] Building CXX object esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifistack.cpp.obj
FAILED: esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifistack.cpp.obj 
ccache /home/feedc0de/.espressif/tools/xtensa-esp32-elf/esp-2022r1-RC1-11.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++ -DHAS_UNCAUGHT_EXCEPTIONS=1 -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -I/home/feedc0de/<-removed->/build/config -I/home/feedc0de/<-removed->/components/espwifistack/src -I/home/feedc0de/<-removed->/esp-idf/components/newlib/platform_include -I/home/feedc0de/<-removed->/esp-idf/components/freertos/FreeRTOS-Kernel/include -I/home/feedc0de/<-removed->/esp-idf/components/freertos/esp_additions/include/freertos -I/home/feedc0de/<-removed->/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -I/home/feedc0de/<-removed->/esp-idf/components/freertos/esp_additions/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/soc -I/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/port/esp32/. -I/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/port/esp32/private_include -I/home/feedc0de/<-removed->/esp-idf/components/heap/include -I/home/feedc0de/<-removed->/esp-idf/components/log/include -I/home/feedc0de/<-removed->/esp-idf/components/soc/include -I/home/feedc0de/<-removed->/esp-idf/components/soc/esp32/. -I/home/feedc0de/<-removed->/esp-idf/components/soc/esp32/include -I/home/feedc0de/<-removed->/esp-idf/components/hal/esp32/include -I/home/feedc0de/<-removed->/esp-idf/components/hal/include -I/home/feedc0de/<-removed->/esp-idf/components/hal/platform_port/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_rom/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_rom/include/esp32 -I/home/feedc0de/<-removed->/esp-idf/components/esp_rom/esp32 -I/home/feedc0de/<-removed->/esp-idf/components/esp_common/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_system/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_system/port/soc -I/home/feedc0de/<-removed->/esp-idf/components/esp_system/port/include/private -I/home/feedc0de/<-removed->/esp-idf/components/xtensa/include -I/home/feedc0de/<-removed->/esp-idf/components/xtensa/esp32/include -I/home/feedc0de/<-removed->/esp-idf/components/lwip/include -I/home/feedc0de/<-removed->/esp-idf/components/lwip/include/apps -I/home/feedc0de/<-removed->/esp-idf/components/lwip/include/apps/sntp -I/home/feedc0de/<-removed->/esp-idf/components/lwip/lwip/src/include -I/home/feedc0de/<-removed->/esp-idf/components/lwip/port/esp32/include -I/home/feedc0de/<-removed->/esp-idf/components/lwip/port/esp32/include/arch -I/home/feedc0de/<-removed->/esp-idf/components/vfs/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_wifi/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_event/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_netif/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_phy/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_phy/esp32/include -I/home/feedc0de/<-removed->/components/cpputils/src -I/home/feedc0de/<-removed->/components/expected/include -I/home/feedc0de/<-removed->/components/fmt/include -I/home/feedc0de/<-removed->/components/espchrono/src -I/home/feedc0de/<-removed->/esp-idf/components/esp_timer/include -I/home/feedc0de/<-removed->/components/date/include -I/home/feedc0de/<-removed->/components/espcpputils/src -I/home/feedc0de/<-removed->/esp-protocols/components/esp_websocket_client/include -I/home/feedc0de/<-removed->/esp-idf/components/esp-tls -I/home/feedc0de/<-removed->/esp-idf/components/esp-tls/esp-tls-crypto -I/home/feedc0de/<-removed->/esp-idf/components/mbedtls/port/include -I/home/feedc0de/<-removed->/esp-idf/components/mbedtls/mbedtls/include -I/home/feedc0de/<-removed->/esp-idf/components/mbedtls/mbedtls/library -I/home/feedc0de/<-removed->/esp-idf/components/mbedtls/esp_crt_bundle/include -I/home/feedc0de/<-removed->/esp-idf/components/driver/include -I/home/feedc0de/<-removed->/esp-idf/components/driver/deprecated -I/home/feedc0de/<-removed->/esp-idf/components/driver/esp32/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_pm/include -I/home/feedc0de/<-removed->/esp-idf/components/esp_ringbuf/include -I/home/feedc0de/<-removed->/esp-idf/components/tcp_transport/include -I/home/feedc0de/<-removed->/esp-idf/components/http_parser -mlongcalls -Wno-frame-address  -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-error=format= -Wno-format -Wno-enum-conversion -gdwarf-4 -ggdb -O2 -fmacro-prefix-map=/home/feedc0de/<-removed->=. -fmacro-prefix-map=/home/feedc0de/<-removed->/esp-idf=/IDF -fstrict-volatile-bitfields -Wno-error=unused-but-set-variable -fno-jump-tables -fno-tree-switch-conversion -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu++11 -fno-exceptions -fno-rtti -D_GNU_SOURCE -DIDF_VER=\"v5.0-dev-3664-g70b4bfab27\" -DESP_PLATFORM -D_POSIX_READER_WRITER_LOCKS -fstack-reuse=all -fstack-protector-all -Wno-unused-function -Wno-deprecated-declarations -Wno-missing-field-initializers -Wno-parentheses -std=gnu++23 -MD -MT esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifistack.cpp.obj -MF esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifistack.cpp.obj.d -o esp-idf/espwifistack/CMakeFiles/__idf_espwifistack.dir/src/espwifistack.cpp.obj -c /home/feedc0de/<-removed->/components/espwifistack/src/espwifistack.cpp
In file included from /home/feedc0de/<-removed->/components/espwifistack/src/espwifistack.cpp:16:
/home/feedc0de/<-removed->/esp-idf/components/esp_system/include/esp_debug_helpers.h: In function 'esp_err_t esp_set_watchpoint(int, void*, int, int)':
/home/feedc0de/<-removed->/esp-idf/components/esp_system/include/esp_debug_helpers.h:118:50: error: invalid conversion from 'int' to 'esp_cpu_watchpoint_trigger_t' [-fpermissive]
  118 |     return esp_cpu_set_watchpoint(no, adr, size, flags);
      |                                                  ^~~~~
      |                                                  |
      |                                                  int
In file included from /home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/hal/cpu_ll.h:18,
                 from /home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/hal/cpu_hal.h:17,
                 from /home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/spinlock.h:11,
                 from /home/feedc0de/<-removed->/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include/freertos/portmacro.h:74,
                 from /home/feedc0de/<-removed->/esp-idf/components/freertos/FreeRTOS-Kernel/include/freertos/portable.h:58,
                 from /home/feedc0de/<-removed->/esp-idf/components/freertos/FreeRTOS-Kernel/include/freertos/FreeRTOS.h:70,
                 from /home/feedc0de/<-removed->/esp-idf/components/esp_event/include/esp_event.h:12,
                 from /home/feedc0de/<-removed->/esp-idf/components/esp_wifi/include/esp_wifi.h:56,
                 from /home/feedc0de/<-removed->/components/espwifistack/src/espwifistack.h:12,
                 from /home/feedc0de/<-removed->/components/espwifistack/src/espwifistack.cpp:1:
/home/feedc0de/<-removed->/esp-idf/components/esp_hw_support/include/esp_cpu.h:475:109: note:   initializing argument 4 of 'esp_err_t esp_cpu_set_watchpoint(int, const void*, size_t, esp_cpu_watchpoint_trigger_t)'
  475 | esp_err_t esp_cpu_set_watchpoint(int wp_num, const void *wp_addr, size_t size, esp_cpu_watchpoint_trigger_t trigger);
      |                                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
ninja: build stopped: subcommand failed.
```

--- a/components/esp_system/include/esp_debug_helpers.h
+++ b/components/esp_system/include/esp_debug_helpers.h
@@ -115,7 +115,7 @@ esp_err_t esp_backtrace_print(int depth);
  */
 static inline __attribute__((deprecated)) esp_err_t esp_set_watchpoint(int no, void *adr, int size, int flags)
 {
-    return esp_cpu_set_watchpoint(no, adr, size, flags);
+    return esp_cpu_set_watchpoint(no, adr, size, (esp_cpu_watchpoint_trigger_t)flags);
 }
 
 /**

